lib/pluginOverride.js

```diff
@@ -66,7 +66,7 @@ module.exports = function override (old, fn, opts) {
     instance[kFourOhFour].arrange404(instance)
   }
 
-  for (const hook of instance[kHooks].onRegister) hook.call(this, instance, opts)
+  for (const hook of instance[kHooks].onRegister) hook.call(instance, instance, opts)
```

Comment:
Isnt this the thing @Eomm meant?

Reply:
I observe no practical difference, nor did the tests, but I can revert it if it's indeed problematic

It's the only `this` usage in that part and forcing instance's usage there seems cleaner to me

Reply:
I would say we should wait for @Eomm 

But I think he wrote, that you have to consider encapsulation. So in onRegister `this` was the parent instance and instance was the new child instance.

Reply:
Actually I looked more and it might be a bug but `this` there is not the parent Fastify instance — the `old` parameter in that function corresponds the to the parent

That function is assigned to an avvio instance so `this` could be that, but I'm not sure what the use of it is

Reply:
Please see a76f91a04bd8c0a44dbadd564dc0934aebdf74b0 to see what I mean

Reply:
Please take a look at a76f91a04bd8c0a44dbadd564dc0934aebdf74b0 to see what I mean